SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/HoloNetViewModel.cs

```diff
@@ -0,0 +1,97 @@
+    {
+        public const int MaxHoloNetTextLength = 800;
+
+        protected override void Initialize(GuiPayloadBase initialPayload)
+        {
+            HoloNetText = string.Empty;
+            WatchOnClient(model => model.HoloNetText);
+        }
+        public string HoloNetText
+        {
+            get => Get<string>();
+            set
+            {
+                Set(value);
+            }
+        }
+
+        public Action OnClickSubmit() => () =>
+        {
+            if (string.IsNullOrWhiteSpace(HoloNetText))
+            {
+                return;
+            }
+
+            var message = HoloNetText;
+
+            if (message.Length > 600)
+            {
+                SendMessageToPC(Player, "Your HoloNet broadcast was too long. Please shorten it to no longer than 600 characters and resubmit the broadcast. For reference, your message was: \"" + message + "\"");
+                return;
+            }
+            var area = GetArea(Player);
+            var position = GetPosition(Player);
+
+            var url = Environment.GetEnvironmentVariable("SWLOR_HOLONET_WEBHOOK_URL");
+
+            if (string.IsNullOrWhiteSpace(url))
+            {
+                SendMessageToPC(Player, ColorToken.Red("ERROR: Unable to send the HoloNet broadcast because server admin has not specified the 'SWLOR_HOLONET_WEBHOOK_HOST' and 'SWLOR_HOLONET_WEBHOOK_PATH' environment variables."));
+                return;
+            }
+
+            var price = 100;
+
+            if (GetGold(Player) < price)
+            {
+                SendMessageToPC(Player, ColorToken.Red("Insufficient credits to make this HoloNet broadcast."));
+                return;
+            }
+
+            else AssignCommand(Player, () => TakeGoldFromCreature(price, Player, true));
+
+            var authorName = $"{GetName(Player)}";
+
+            Task.Run(async () =>
+            {
+                using (var client = new DiscordWebhookClient(url))
+                {
+                    var embed = new EmbedBuilder
+                    {
+                        Description = message,
+                        Author = new EmbedAuthorBuilder
+                        {
+                            Name = authorName
+                        },
+                        Color = Color.Blue
+                    };
+
+                    await client.SendMessageAsync(string.Empty, embeds: new[] { embed.Build() });
+                }
+            });
+
+            SendMessageToPC(Player, "HoloNet message broadcasted!");
+            Gui.TogglePlayerWindow(Player, GuiWindowType.HoloNet);
+
+            for (var onlinePlayer = GetFirstPC(); GetIsObjectValid(onlinePlayer); onlinePlayer = GetNextPC())
+            {
+                   SendMessageToPC(onlinePlayer, ColorToken.Custom("A new HoloNet message has been broadcasted by " + authorName + ": ", 0, 180, 255) + ColorToken.White(message));
```

Comment:
You have an extra tab here that should be removed.

Reply:
I would also shorten the message to something like:

$"{authorName} broadcasts a message: "